secure_software_development_fundamentals.md

```diff
@@ -2488,6 +2488,62 @@ explained in the [PostgreSQL (Command Execution Functions) documentation](https:
 
 The [OWASP Query Parameterization Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Query_Parameterization_Cheat_Sheet.html) and [Bobby Tables website](https://bobby-tables.com/) provide examples for a variety of ecosystems.
 
+##### Subtle issues: DBMS (Server) side vs. Application (client) side
+
+An important security issue is *where* the
+parameters of a parameterized statement are processed.
+There are two options, DBMS-side and application-side, and
+DBMS-side is better from a security point of view.
+
+From a security point-of-view it's best if the parameters of
+parameterized statements are be processed directly
+within the database management system (DBMS),
+aka "DBMS-side" parameter processing.
+This approach is often called "server-side" since many DBMSs use
+client/server architectures connecting over a network.
+There are many advantages to DBMS-side parameter processing.
+The DBMS has the current information on escaping rules
+(and can often use more efficient mechanisms than adding escape characters),
```

Comment:
More efficient or also more effective?

Reply:
Efficient. "Effective" is covered by the text before the parentheses.